content/guides/spring-integration-lock/index.md

```diff
@@ -0,0 +1,353 @@
+---
+date: '2021-10-22'
+lastmod: '2021-10-22'
+title: Working With the Spring Distributed Lock
+linkTitle: Working With the Spring Distributed Lock
+description: A simple demonstration of how to implement a Spring Distributed Lock with in your Spring application
+patterns:
+  - API
+tags:
+  - Spring Boot
+  - Spring Integration
+  - API
+  - Getting Started
+  - Spring
+  - Microservices
+team:
+  - Eric Standley
+weight: 2
+oldPath: '/content/guides/spring/spring-distributed-lock.md'
+aliases:
+  - '/guides/spring/spring-distributed-lock'
+level1: Building Modern Applications
+level2: Frameworks and Languages
+---
+
+When designing microservices, the typical pattern is to design services that can handle multiple instances running at the same time. However, there are situations where only a single service is preferable. For example, in the [Leader Pattern](https://docs.microsoft.com/en-us/azure/architecture/patterns/leader-election), you cannot synchronize if the code runs in different pods. As a result, you have to use an external method that is fraught with pitfalls during implementation[^1]. 
```

Comment:
`However, there are situations where only a single service is preferable.`  Isn't totally the correct idea, would `However, there are situations where processing only on a single service is preferable.` work?

Reply:
Seems okay to me. @lizz7000?